src/configure/clients/modaTemplateServiceClient.ts

```diff
@@ -0,0 +1,87 @@
+
+import { RepositoryAnalysis } from "azureintegration-repoanalysis-client-internal";
+import { RestClient } from "typed-rest-client";
+import vscodeUri from "vscode-uri";
+import { ExtendedPipelineTemplate } from "../model/Contracts";
+import { StringMap } from "../model/models";
+import { TemplateInfo } from "../model/templateModels";
+import { ITemplateServiceClient } from "./ITemplateServiceClient";
+
+export class ModaTemplateServiceClient implements ITemplateServiceClient {
+    private restClient: RestClient;
+    private githubPat: string;
+    private pathUrl: string;
+    private readonly apiVersion = "6.0-preview.1";
+    private readonly extendedPipelineTemplateResource = "ExtendedPipelineTemplates";
+    private readonly templatesInfoResource = "TemplatesInfo";
+    private readonly templateAssetFilesResource = "TemplateAssetFiles";
+
+    constructor(url: string, githubPat: string) {
+        const u = vscodeUri.parse(url);
+        this.restClient = new RestClient("deploy-to-azure", u.scheme + "://" + u.authority);
+        this.pathUrl = u.path;
+        this.githubPat = githubPat;
+    }
+
+    public async getTemplates(body: RepositoryAnalysis): Promise<TemplateInfo[]> {
+        const requestUri = this.pathUrl + this.templatesInfoResource;
+        const requestOptions = {
+            acceptHeader: "application/json",
+            additionalHeaders: {
+                "Authorization": "Bearer " + this.githubPat
```

Comment:
Does `Bearer` work or should it be `token`? I think you should check what is used when sending request for setting github secret.
I know it is really difficult to write accurate code when there is no real API to test. But we can do best guess. Infact, a difficult way to test exists, if you add your IP to whitelisted IPs for moda API. THat's how deepak and tejas test their changes.